curvefs/src/client/fuse_s3_client.cpp

```diff
@@ -112,13 +112,17 @@ bool FuseS3Client::InitKVCache(const KVClientManagerOpt &opt) {
         return false;
     }
 
-    g_kvClientManager = new KVClientManager();
-    if (!g_kvClientManager->Init(opt, memcacheClient)) {
+    kvClientManager_ = std::make_shared<KVClientManager>();
```

Comment:
> I saw you share `kvClientManager_` everywhere, so please rethink its lifetime.
> 
> If you can ensure no one else will use it after `FuseS3Client::UnInit`, then, only FuseS3Client should be the only owner of `kvClientManager_`, all others are just users, so just pass pointer to others

Observing the old code, we can find that g_kvClientManager was created by FuseS3Client during InitKVCache, but there is no delete recovery. InitKVCache is a private function of FuseS3Client, and it is called in FuseS3Client::Init, and it is not actively recycled in FuseS3Client::UnInit, even if it is used, it can still be accessed normally unless it is re-Inited.

Reply:
> I saw you share `kvClientManager_` everywhere, so please rethink its lifetime.
> 
> If you can ensure no one else will use it after `FuseS3Client::UnInit`, then, only FuseS3Client should be the only owner of `kvClientManager_`, all others are just users, so just pass pointer to others

Observing the old code, we can find that g_kvClientManager was created by FuseS3Client during InitKVCache, but there is no delete recovery. InitKVCache is a private function of FuseS3Client and is called in FuseS3Client::Init. And it is not actively recycled in FuseS3Client::UnInit, even if it is used, it can still be accessed normally unless it is re-Inited. This is available throughout its lifetime.